submissions/svart0222/port-sniffer/index.js

```diff
@@ -0,0 +1,75 @@
+const net = require('net');
+
+const checkPort = (port, host) => {
+  return new Promise(resolve => {
+    const socket = net.createConnection(port, host, () => {
+      clearTimeout(timer);
+      resolve({ port, result: true });
+      socket.end();
+    });
+    const timer = setTimeout(() => {
+      resolve({ port, result: false });
+      socket.end();
+    }, 300);
+    socket.on('error', () => {
+      clearTimeout(timer);
+      resolve({ port, result: false });
+    });
+  });
+};
+
+const getResult = async (host, startPoint = 0, endPoint = 65535) => {
+  try {
+    const result = [];
+    let currentPoint = startPoint;
+
+    while (currentPoint < endPoint) {
+      result.push(checkPort(currentPoint, host));
+      currentPoint += 1;
+    }
+    const scanResult = await Promise.all(result);
+    scanResult.forEach(el => {
+      if (el.result) process.stdout.write(`Port ${el.port} is opened\n`);
+    });
+  } catch (e) {
+    process.stdout.write(e);
```

Comment:
better use console.log

Reply:
but if we will use console we can't see result in the terminal (if call from terminal, without debugger), am I right ?